src/ng/compile.js

```diff
@@ -2129,6 +2139,49 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
     /**
+     * Matches elements in the transcluded content to elements in the directive template by ng-transclude-selector.
+     * Used in directives that set transclude to 'multi'.
+     *
+     * @param {jqLite} dirElement Main directive element
+     * @param {function()} transcludeFn Transclusion function for the directive
+    **/
+    function multiTransclude(dirElement, transcludeFn) {
+      transcludeFn(transcludeCallback);
+
+      function transcludeCallback(clone) {
+        var target,
+            selector,
+            selectedElements,
+            transcludeTargets = dirElement[0].querySelectorAll('[ng-transclude-select]'),
+            cloneWrapper = jqLite("<span></span>");
+        cloneWrapper.append(clone);
```

Comment:
This is definitely code smell, but I couldn't figure out a better way to search the clone by selector.  Any suggestions on how to improve?